pkg/services/ngalert/store/image.go

```diff
@@ -36,6 +40,19 @@ func (st DBstore) GetImage(ctx context.Context, token string) (*models.Image, er
 	return &img, nil
 }
 
+func (st DBstore) GetImages(ctx context.Context, tokens []string) ([]models.Image, error) {
+	var imgs []models.Image
```

Comment:
The use we have in mind for this is inside `withStoredImages`, which guarantees that it will call `imageFunc` for every alert in the list, whether there's a result or not. If there's no result, the image will be `nil`

One way we could support that is by returning a `map[string]*Image` here from token to image. Another approach could be an ordered `[]*Image` with nil entries, in the same order as the args. 
We could also have `withStoredImages` do that processing after the GetImages call, too.

Reply:
I had planned for `withStoredImages` to do indexing where required 👍